pkg/mock/handlers/handlers.go

```diff
@@ -0,0 +1,162 @@
+		routes = trimmedRoutes
+	} else if strings.HasPrefix(trimmedRoute, dynamic.ServicePath) {
+		trimmedRoute = strings.TrimPrefix(trimmedRoute, dynamic.ServicePath+"/")
+		log.Println("dynamic prefix detected..trimming: ", trimmedRoute)
+		routes = trimmedRoutesDynamic
+	} else {
+		server.ReturnNotFoundResponse(res)
+		return
+	}
+
+	if paths, ok := routeLookupTable[trimmedRoute]; ok {
+		log.Printf("CatchAllHandler entry %s already in map: %v \n", trimmedRoute, routeLookupTable[trimmedRoute])
+		server.FormatAndReturnTextResponse(res, strings.Join(paths, "\n"))
+		return
+	}
+
+	appendToResults := true
+	for _, route := range routes {
+		if strings.Contains(route, trimmedRoute) {
+			log.Printf("route: %s   contains trimmed route: %s \n", route, trimmedRoute)
+			// ex: iam/security-credentials --> security-credentials
+			route = strings.TrimPrefix(route, trimmedRoute+"/")
+			for i, existingRoute := range results {
+				// ex: security-credentials/baskinc-role contains security-credentials
+				if strings.Contains(route, existingRoute) {
+					log.Printf("route: %s   contains existingRoute: %s \n", route, existingRoute)
+					// display as security-credentials/
+					results[i] = existingRoute + "/"
+					log.Printf("updated existingRoute: %s \n", existingRoute)
+					appendToResults = false
+					break
+				}
+			}
+			if appendToResults {
+				log.Printf("adding route: %s to results\n", route)
+				results = append(results, route)
+			}
+			appendToResults = true
+		}
+	}
+
+	if len(results) <= 0 {
+		server.ReturnNotFoundResponse(res)
+		return
+	}
+
+	routeLookupTable[trimmedRoute] = results
+	log.Printf("CatchAllHandler: adding  %s  and its routes: %v to the map\n", trimmedRoute, results)
+	server.FormatAndReturnTextResponse(res, strings.Join(results, "\n"))
+	return
+}
+
+// ListRoutesHandler returns the list of supported paths
+func ListRoutesHandler(res http.ResponseWriter, req *http.Request) {
+	log.Println("Received request to display paths: ", req.URL.Path)
+	// Routes are not available until runtime; only want to do this ONCE
+	if len(trimmedRoutes) == 0 {
+		formatRoutes()
+	}
+
+	// these paths do not use routeLookupTable due to inconsistency of trailing "/" with IMDS
+	switch req.URL.Path {
+	case static.ServicePath:
+		server.FormatAndReturnTextResponse(res, strings.Join(trimmedRoutes, "\n")+"\n")
+	case dynamic.ServicePath:
+		server.FormatAndReturnTextResponse(res, strings.Join(trimmedRoutesDynamic, "\n")+"\n")
+	case latestPath:
+		server.FormatAndReturnTextResponse(res, strings.Join(supportedCategories, "\n")+"\n")
+	case versionsPath:
+		server.FormatAndReturnTextResponse(res, strings.Join(supportedVersions, "\n")+"\n")
+	default:
+		server.ReturnNotFoundResponse(res)
+	}
+
+	return
+}
+
+func formatRoutes() {
+	var trimmedRoute string
+	for _, route := range server.Routes {
+
+		if strings.HasPrefix(route, dynamic.ServicePath) {
+			// Omit /latest/dynamic
```

Comment:
ah I remember now. When the path: `/latest/meta-data` enters this code if I try to trim the prefix all at once, then it won't succeed and the path will be appended to `trimmedRoutes`. I don't want this because then the path will be listed when querying `localhost/latest/meta-data`:

```
/latest/meta-data
ami-id
ami-launch-index
ami-manifest-path
```